### DISCUSSION TITLE
feat(tracing): perfetto exporter

### DISCUSSION THREAD

**Discussion by bsarden** [open-telemetry/opentelemetry-cpp#1483] [General]
I'm evaluating the differences between perfetto and the open-telemetry ecosystem, and it seems like there could be a lot of value in reusing the perfetto ui for viewing otel metrics+span sdk's. Specifically, being able to reuse the [trace visualization](https://perfetto.dev/docs/#trace-visualization) and the [trace processor](https://perfetto.dev/docs/analysis/trace-processor) to have a cohesive story for running SQL expressions on traces. 

Besides otel lacking the profiling support, being able to reuse the span / metrics sdk as a drop-in replacement for perfetto's `TRACE_EVENT` and `TRACE_COUNTER` macros seems like it may be worthwhile to others as well. Is this something other members in the community are interested in as well?

Naively, it seems like perfetto was designed for tracing on-device behavior while otel focuses more on distributed tracing currently.  Maybe there is no overlap here, but figured it would be worth asking about :).

**Comment by lalitb** [comment]
@bsarden  -Looks like this is a more suitable discussion for the specs repo (https://github.com/open-telemetry/opentelemetry-specification). This repo is for OpenTelemetry C++.

**Comment by cforce** [comment]
"Tracing on-device behavior" suggests that if the SDK or client generates OTLP-compliant messages, it could integrate into the OpenTelemetry ecosystem. This would allow it to leverage some of the features not available in the OpenTelemetry Collector's host metrics receiver, while offering performance similar to eBPF through Linux tracing.
related eBPF based tooling https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34321

**Comment by cforce** [comment]
related https://github.com/open-telemetry/opentelemetry-android/issues/655
